### DISCUSSION TITLE
Lesson 7, Part 19 - Interactions.s.sol and InteractionsTest.t.sol - undeclared identifier - is there a problem with the code of foundry_devops?

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#546] [General]
I have the same code for Interactions.s.sol and InteractionsTest.t.sol as Patrick but I was getting undeclared identifier errors relating to the use of "fundMe" throughout InteractionsTest.t.sol when I tried to run "forge test" or "forge script script/Interactions.s.sol:FundFundMe"...and yes I did have a state variable "Fund Me fundMe".

I decided to try some troubleshooting...first I ran "forge script script/DeployFundMe.s.sol" and that worked fine. Then I tried putting "forge script script/Interactions.s.sol:FundFundMe --fork-url $SEPOLIA_RPC_URL" into the terminal to see if it made any difference and I got a bunch of errors (see screencap below) that seem to relate to Patrick's foundry_devops code. Does anyone know if there are errors in that code? For reference I am pasting the relevant code below...let me know if you know how to fix any of the errors in the screencap. I also tried running it as "forge script script/Interactions.s.sol:FundFundMe --rpc-url $SEPOLIA_RPC_URL --private-key $SEPOLIA_PRIVATE_KEY --broadcast" (using source .env) and I got the same errors.

get_recent_deployment code:

`#!/bin/bash

# Get the script's own directory
SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"

# Move up two levels to get the project root directory
PROJECT_ROOT_DIR="$(dirname "$(dirname "$SCRIPT_DIR")")"

contractName=$1
chainId=$2

if [ -z "$3" ]; then
  path="./broadcast"
else
  path="$3"
fi

if [ -z "$contractName" ] || [ -z "$chainId" ] ; then
  echo "Usage: $0 <contractName> <chainId> [path]"
  exit 1
fi

latestTimestamp=0
latestContractAddress=""

files=$(find $path -name "run-latest.json" -type f)

for file in $files; do
  if [[ $file == *"/$chainId/"* ]]; then
    timestamp=$(jq '.timestamp' "$file")
    
    transactions_length=$(jq '.transactions | length' "$file")

    for ((i=0; i<$transactions_length; i++)); do
      currentTransactionType=$(jq -r ".transactions[$i].transactionType" "$file")
      currentContractName=$(jq -r ".transactions[$i].contractName" "$file")

      if [ "$currentTransactionType" == "CREATE" ] && [ "$currentContractName" == "$contractName" ] && [ $timestamp -gt $latestTimestamp ]; then
        latestTimestamp=$timestamp
        latestContractAddress=$(jq -r ".transactions[$i].contractAddress" "$file")
      fi
    done
  fi
done

if [ $latestTimestamp -ne 0 ]; then
  echo "$latestContractAddress"
else
  echo "0x0000000000000000000000000000000000000000"
fi
`

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/138697105/391d84fa-6a26-454c-87c6-db73020c40b5)

**Comment by shikhar229169** [comment]
You need to install jq
You can install it by:
```
sudo apt install jq
```
